tests/untldoc_test.py

```diff
@@ -0,0 +1,689 @@
+               '  </creator>\n'
+               '  <publisher>\n'
+               '    <name>Fake Publishing</name>\n'
+               '  </publisher>\n'
+               '  <date qualifier="creation">1944</date>\n'
+               '  <collection>UNT</collection>\n'
+               '</metadata>\n')
+
+DC_DICTIONARY = {'title': [{'content': 'Tres Actos'}],
+                 'creator': [{'content': 'Enhorn, Blair'}],
+                 'publisher': [{'content': 'Fake Publishing'}],
+                 'date': [{'content': '1944'}]}
+
+
+def test_PyuntlException():
+    msg = 'Throw this'
+    try:
+        raise untldoc.PyuntlException(msg)
+    except untldoc.PyuntlException as err:
+        assert str(err) == msg
+
+
+def test_untlxml2py():
+    """Verify children are correctly added from the UNTL XML."""
+    xml = BytesIO(b'<?xml version="1.0" encoding="UTF-8"?>\n'
+                  b'<metadata>\n'
+                  b'  <title qualifier="officialtitle">Tres Actos</title>\n'
+                  b'  <creator qualifier="aut">\n'
+                  b'    <name>Last, Furston, 1807-1865.</name>\n'
+                  b'  </creator>\n'
+                  b'</metadata>\n')
+    root = untldoc.untlxml2py(xml)
+    assert isinstance(root, us.UNTLElement)
+    assert root.tag == 'metadata'
+    assert len(root.children) == 2
+    assert root.children[0].tag == 'title'
+    assert root.children[0].content == 'Tres Actos'
+    assert root.children[0].qualifier == 'officialtitle'
+    assert root.children[1].tag == 'creator'
+    assert root.children[1].qualifier == 'aut'
+    assert root.children[1].children[0].tag == 'name'
+    assert root.children[1].children[0].content == 'Last, Furston, 1807-1865.'
+
+
+def test_untlxml2py_namespace():
+    """Namespace is removed from element.tag when building py object.
+
+    Including xmlns in the xml prepends "{http://purl.org/dc/elements/1.1/}"
+    to element tags, but untlxml2py will ignore that.
+    """
+    xml = BytesIO(b'<?xml version="1.0" encoding="UTF-8"?>\n'
+                  b'<metadata xmlns="http://purl.org/dc/elements/1.1/">\n'
+                  b'  <title qualifier="officialtitle">Tres Actos</title>\n'
+                  b'</metadata>\n')
+    root = untldoc.untlxml2py(xml)
+    assert isinstance(root, us.UNTLElement)
+    assert root.tag == 'metadata'
+    assert root.children[0].tag == 'title'
+
+
+def test_untlxml2py_non_UNTL_tag_raises_exception():
+    xml = BytesIO(b'<?xml version="1.0" encoding="UTF-8"?>\n'
+                  b'<metadata>\n'
+                  b'  <dog>Bezos</dog>\n'
+                  b'</metadata>\n')
+    with pytest.raises(untldoc.PyuntlException) as err:
+        untldoc.untlxml2py(xml)
+    assert 'Element "dog" not in UNTL dispatch.' == err.value.args[0]
+
+
+def test_untlxml2pydict():
+    xml = BytesIO(UNTL_STRING.encode('utf-8'))
+    untl_dict = untldoc.untlxml2pydict(xml)
+    assert untl_dict == UNTL_DICTIONARY
+
+
+def test_untlpydict2xml(tmpdir):
```

Comment:
I go back and forth on testing functions like this--are we testing that the function is a wrapper around some function in another module or are we testing the result of running the function under test...

Reply:
One could also ask whether such functions are worth having at all if they only server to call another function which could have just been called directly...

Reply:
But _if_ we retain _and_ test functions like this, then my personal opinion is that we should only be testing what's directly inside the function under test. Any other functions that are called will presumably have their own tests. To me, if the purpose of a function is "to call another function", then I wouldn't bother testing _what_ that other function does, just that it does indeed get called.

Reply:
Yeah...

Regarding simply checking it calls the function, by the time you set up a mock and inputs to call the mock with so that you can then verify the mocked function was called appropriately and that the return value of the function under test is the same return value of the mocked function, it seems like it is not so simple...

For consistency with the other tests you have brought up that I don't want to change now, I am going to test without mocking--looking at the expected output of the function instead.

Reply:
Okie dokie, that'll work